src/metrics/metrics_test.go

```diff
@@ -499,27 +474,151 @@ func TestPopulatePgBouncerMetrics(t *testing.T) {
 		"host":        "testhost",
 	}
 
-	expectedPool := map[string]interface{}{
-		"pgbouncer.pools.clientConnectionsActive":  float64(1),
-		"pgbouncer.pools.clientConnectionsWaiting": float64(2),
-		"pgbouncer.pools.serverConnectionsActive":  float64(3),
-		"pgbouncer.pools.serverConnectionsIdle":    float64(4),
-		"pgbouncer.pools.serverConnectionsUsed":    float64(5),
-		"pgbouncer.pools.serverConnectionsTested":  float64(6),
-		"pgbouncer.pools.serverConnectionsLogin":   float64(7),
-		"pgbouncer.pools.maxwaitInMilliseconds":    float64(8),
-		"displayName":                              "testDB",
-		"entityName":                               "pgbouncer:testDB",
-		"event_type":                               "PgBouncerSample",
-		"host":                                     "testhost",
+	testsCases := []struct {
+		name string
+		// using a copy since sqlmock gets modified.
+		pgbouncerStatsRows sqlmock.Rows
```

Comment:
Is this needed to be passed on each test case? I think we can use it right in the loop.

Reply:
actually no, i can move it inside the loop

Reply:
done